Obsidian/Utilities/ConsoleIO.cs

```diff
@@ -0,0 +1,380 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace Obsidian.Utilities
+{
+
+    /// <summary>
+    /// Allows simultaneous console input and output without breaking user input
+    /// (Without having this annoying behaviour : User inp[Some Console output]ut)
+    /// Provide some fancy features such as formatted output, text pasting and tab-completion.
+    /// By ORelio - (c) 2012-2018 - Available under the CDDL-1.0 license
```

Comment:
I'm just wondering, is the license OK? Honestly, I'm not a big fan of how the whole class is implemented, so I might eventually rewrite it to something custom anyways. Seems fine for now.